pkg/controllers/nodeclass/validation_test.go

```diff
@@ -306,4 +309,34 @@ var _ = Describe("NodeClass Validation Status Controller", func() {
 		ExpectNotFound(ctx, env.Client, nodeClass)
 		Expect(awsEnv.ValidationCache.Items()).To(HaveLen(0))
 	})
+	It("should pass validation when the validation controller is disabled", func() {
+		controller = nodeclass.NewController(
+			awsEnv.Clock,
+			env.Client,
+			cloudProvider,
+			events.NewRecorder(&record.FakeRecorder{}),
+			fake.DefaultRegion,
+			awsEnv.SubnetProvider,
+			awsEnv.SecurityGroupProvider,
+			awsEnv.AMIProvider,
+			awsEnv.InstanceProfileProvider,
+			awsEnv.InstanceTypesProvider,
+			awsEnv.LaunchTemplateProvider,
+			awsEnv.CapacityReservationProvider,
+			awsEnv.EC2API,
+			awsEnv.ValidationCache,
+			awsEnv.RecreationCache,
+			awsEnv.AMIResolver,
+			true,
+		)
+		ExpectApplied(ctx, env.Client, nodeClass)
+		ExpectObjectReconciled(ctx, env.Client, controller, nodeClass)
+		nodeClass = ExpectExists(ctx, env.Client, nodeClass)
+		Expect(nodeClass.StatusConditions().Get(v1.ConditionTypeValidationSucceeded).IsTrue()).To(BeTrue())
+		Expect(nodeClass.StatusConditions().Get(status.ConditionReady).IsTrue()).To(BeTrue())
+		// The cache shouldn't have any entries because we short circuit
+		Expect(awsEnv.ValidationCache.Items()).To(HaveLen(0))
+		// We shouldn't make any CreateFleet calls when validation is disabled
+		Expect(awsEnv.EC2API.CreateFleetBehavior.Calls()).To(Equal(0))
```

Comment:
Should we also add run instances, instance profiles, and launch templates as well?

Reply:
It would just be run instances and launch templates but yes I can add it.